### ISSUE TITLE
Failed to install AFDKO on Windows 11 22H2

### ISSUE THREAD

**Issue by pkj-l** [adobe-type-tools/afdko#1574]
Hi, I'm trying to install AFDKO on my Windows 11 22H2 laptop, but failed.
My Python version is `3.11`. My Pip version is `22.3`.
```
pip install --user afdko
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Collecting afdko
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/4f/7e/3c2251abada7a47f56508d3177bc76ba4644c9399d0e1ea4fb778ab799a3/afdko-3.9.1-py3-none-win_amd64.whl (4.6 MB)
Collecting lxml>=4.9.0
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/70/bb/7a2c7b4f8f434aa1ee801704bf08f1e53d7b5feba3d5313ab17003477808/lxml-4.9.1.tar.gz (3.4 MB)
  Preparing metadata (setup.py) ... done
Collecting booleanOperations>=0.9.0
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/fc/c6/c4cae54f482465a33c5f011d95ec64293dce9e012dac7873147c2dc85396/booleanOperations-0.9.0-py3-none-any.whl (18 kB)
Collecting defcon[lxml,pens]>=0.10.1
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/09/46/753bd50b5104234f4b4f2bcb5c9790a3e89357a083fefbd566453653c1bc/defcon-0.10.2-py3-none-any.whl (231 kB)
Collecting fontMath>=0.9.2
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/af/ae/11b1d4201107a936bfa936b5010b8840dc14ad81ffc9539a05ef85a8013a/fontMath-0.9.2-py2.py3-none-any.whl (32 kB)
Requirement already satisfied: fontTools[lxml,ufo,unicode,woff]>=4.33.3 in c:\users\poetj\appdata\roaming\python\python311\site-packages (from afdko) (4.38.0)
Collecting psautohint>=2.4.0
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/91/74/014da6e9280844cec6a73a04e069d95740e5520fd9477afbf31208d0e2f4/psautohint-2.4.0.tar.gz (185 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting tqdm>=4.64.0
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/47/bb/849011636c4da2e44f1253cd927cfb20ada4374d8b3a4e425416e84900cc/tqdm-4.64.1-py2.py3-none-any.whl (78 kB)
Requirement already satisfied: ufonormalizer>=0.6.1 in c:\users\poetj\appdata\roaming\python\python311\site-packages (from afdko) (0.6.1)
Collecting ufoProcessor>=1.9.0
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/47/66/8ce437915371d6813339d236d8beb6cbef601d125e94d7c09bfa0f6dd5c6/ufoProcessor-1.9.0-py2.py3-none-any.whl (22 kB)
Requirement already satisfied: pyclipper>=1.1.0.post1 in c:\users\poetj\appdata\roaming\python\python311\site-packages (from booleanOperations>=0.9.0->afdko) (1.3.0.post3)
Collecting fontPens>=0.1.0
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/ba/9d/d084c2760c735e8f81c2a3c9e0d403cfacb811daf5166fbaa353e6d647c5/fontPens-0.2.4-py2.py3-none-any.whl (19 kB)
Collecting fs<3,>=2.2.0
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/b9/5c/a3d95dc1ec6cdeb032d789b552ecc76effa3557ea9186e1566df6aac18df/fs-2.4.16-py2.py3-none-any.whl (135 kB)
Requirement already satisfied: zopfli>=0.1.4 in c:\users\poetj\appdata\roaming\python\python311\site-packages (from fontTools[lxml,ufo,unicode,woff]>=4.33.3->afdko) (0.2.1)
Requirement already satisfied: brotli>=1.0.1 in c:\users\poetj\appdata\roaming\python\python311\site-packages (from fontTools[lxml,ufo,unicode,woff]>=4.33.3->afdko) (1.0.9)
Collecting colorama
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/d1/d6/3965ed04c63042e047cb6a3e6ed1a63a35087b6a609aa3a15ed8ac56c221/colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting fontParts>=0.8.2
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/b4/7c/13d7db92e61d597907564ce9b715bc5fad8d5b1dee7215e066182110ad14/fontParts-0.10.8-py3-none-any.whl (165 kB)
Collecting mutatorMath>=2.1.2
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/02/61/e3e6190a8d94c6a99d9605c721e1e38ebae19dc194059477affa57f6e111/MutatorMath-3.0.1-py2.py3-none-any.whl (31 kB)
Requirement already satisfied: appdirs~=1.4.3 in c:\users\poetj\appdata\roaming\python\python311\site-packages (from fs<3,>=2.2.0->fontTools[lxml,ufo,unicode,woff]>=4.33.3->afdko) (1.4.4)
Requirement already satisfied: setuptools in c:\users\poetj\appdata\local\programs\python\python311\lib\site-packages (from fs<3,>=2.2.0->fontTools[lxml,ufo,unicode,woff]>=4.33.3->afdko) (65.5.0)
Requirement already satisfied: six~=1.10 in c:\users\poetj\appdata\roaming\python\python311\site-packages (from fs<3,>=2.2.0->fontTools[lxml,ufo,unicode,woff]>=4.33.3->afdko) (1.16.0)
Building wheels for collected packages: psautohint
  Building wheel for psautohint (pyproject.toml) ... done
  Created wheel for psautohint: filename=psautohint-2.4.0-cp311-cp311-win_amd64.whl size=136087 sha256=f737cf21e44feaabfe03036d17d97e96b100f00877384fdc900b2dedbe6a3004
  Stored in directory: c:\users\poetj\appdata\local\pip\cache\wheels\71\77\8f\3134e89b4b94ac9ac3044839aa9aaf2c2e317b0e7d00ab5172
Successfully built psautohint
Installing collected packages: lxml, fs, fontPens, fontMath, colorama, booleanOperations, tqdm, psautohint, defcon, mutatorMath, fontParts, ufoProcessor, afdko
  DEPRECATION: lxml is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for lxml ... error
  error: subprocess-exited-with-error

  × Running setup.py install for lxml did not run successfully.
  │ exit code: 1
  ╰─> [96 lines of output]
      Building lxml version 4.9.1.
      Building without Cython.
      Building against pre-built libxml2 andl libxslt libraries
      running install
      C:\Users\PoetJ\AppData\Local\Programs\Python\Python311\Lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\builder.py -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\cssselect.py -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\doctestcompare.py -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\ElementInclude.py -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\pyclasslookup.py -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\sax.py -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\usedoctest.py -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\_elementpath.py -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\__init__.py -> build\lib.win-amd64-cpython-311\lxml
      creating build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\__init__.py -> build\lib.win-amd64-cpython-311\lxml\includes
      creating build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\builder.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\clean.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\defs.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\diff.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\ElementSoup.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\formfill.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\html5parser.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\soupparser.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\usedoctest.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\_diffcommand.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\_html5builder.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\_setmixin.py -> build\lib.win-amd64-cpython-311\lxml\html
      copying src\lxml\html\__init__.py -> build\lib.win-amd64-cpython-311\lxml\html
      creating build\lib.win-amd64-cpython-311\lxml\isoschematron
      copying src\lxml\isoschematron\__init__.py -> build\lib.win-amd64-cpython-311\lxml\isoschematron
      copying src\lxml\etree.h -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\etree_api.h -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\lxml.etree.h -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\lxml.etree_api.h -> build\lib.win-amd64-cpython-311\lxml
      copying src\lxml\includes\c14n.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\config.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\dtdvalid.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\etreepublic.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\htmlparser.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\relaxng.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\schematron.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\tree.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\uri.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\xinclude.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\xmlerror.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\xmlparser.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\xmlschema.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\xpath.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\xslt.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\__init__.pxd -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\etree_defs.h -> build\lib.win-amd64-cpython-311\lxml\includes
      copying src\lxml\includes\lxml-version.h -> build\lib.win-amd64-cpython-311\lxml\includes
      creating build\lib.win-amd64-cpython-311\lxml\isoschematron\resources
      creating build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\rng
      copying src\lxml\isoschematron\resources\rng\iso-schematron.rng -> build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\rng
      creating build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\xsl
      copying src\lxml\isoschematron\resources\xsl\RNG2Schtrn.xsl -> build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\xsl
      copying src\lxml\isoschematron\resources\xsl\XSD2Schtrn.xsl -> build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\xsl
      creating build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
      copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\iso_abstract_expand.xsl -> build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
      copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\iso_dsdl_include.xsl -> build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
      copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\iso_schematron_message.xsl -> build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
      copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\iso_schematron_skeleton_for_xslt1.xsl -> build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
      copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\iso_svrl_for_xslt1.xsl -> build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
      copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\readme.txt -> build\lib.win-amd64-cpython-311\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
      running build_ext
      building 'lxml.etree' extension
      creating build\temp.win-amd64-cpython-311
      creating build\temp.win-amd64-cpython-311\Release
      creating build\temp.win-amd64-cpython-311\Release\src
      creating build\temp.win-amd64-cpython-311\Release\src\lxml
      "C:\Program Files\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.33.31629\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -DCYTHON_CLINE_IN_TRACEBACK=0 -Isrc -Isrc\lxml\includes -IC:\Users\PoetJ\AppData\Local\Programs\Python\Python311\include -IC:\Users\PoetJ\AppData\Local\Programs\Python\Python311\Include "-IC:\Program Files\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.33.31629\include" "-IC:\Program Files\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt" /Tcsrc\lxml\etree.c /Fobuild\temp.win-amd64-cpython-311\Release\src\lxml\etree.obj -w
      cl: 命令行 warning D9025 :正在重写“/W3”(用“/w”)
      etree.c
      C:\Users\PoetJ\AppData\Local\Temp\pip-install-164_5t29\lxml_66257dafd8aa44fc9e5e88170df88a17\src\lxml\includes/etree_defs.h(14): fatal error C1083: 无法打开包括文件: “libxml/xmlversion.h”: No such file or directory
      Compile failed: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.33.31629\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      creating Users
      creating Users\PoetJ
      creating Users\PoetJ\AppData
      creating Users\PoetJ\AppData\Local
      creating Users\PoetJ\AppData\Local\Temp
      "C:\Program Files\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.33.31629\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -I/usr/include/libxml2 "-IC:\Program Files\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.33.31629\include" "-IC:\Program Files\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt" /TcC:\Users\PoetJ\AppData\Local\Temp\xmlXPathInitscgojqbw.c /FoUsers\PoetJ\AppData\Local\Temp\xmlXPathInitscgojqbw.obj
      xmlXPathInitscgojqbw.c
      C:\Users\PoetJ\AppData\Local\Temp\xmlXPathInitscgojqbw.c(1): fatal error C1083: 无法打开包括文件: “libxml/xpath.h”: No such file or directory
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.33.31629\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      *********************************************************************************
      Could not find function xmlCheckVersion in library libxml2. Is libxml2 installed?
      *********************************************************************************
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> lxml

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Comment by skef** [comment]
Looks like the problem is with a dependency (the `lxml` python package) and the fact that wheels aren't yet available for Python 3.11 on Windows. They're aware of the problem: https://bugs.launchpad.net/lxml/+bug/1977998

I would keep an eye on that issue and try again when it's resolved.

**Comment by skef** [comment]
(If you want to work around this, try moving back from Python 3.11 to Python 3.10. 3.11 is new enough to count as "bleeding edge" at present.)
